compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1644,6 +1657,9 @@ private ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLook
                 return ParserRuleContext.EXPRESSION_RHS;
             case LOCAL_TYPE_DEFINITION_STMT:
                 return ParserRuleContext.TYPE_KEYWORD;
+            case NULL_KEYWORD:
+                endContext(); // end nil-literal
```

Comment:
A context is not started for null keyword.. So ending the context here is not correct. 

We might not need any context for null keyword at all. It is very similar to string literal, int literal etc. A context is needed when a token can occur in different contexts/places  and if we can't determine whats next rule should be.. Then we make use of the parent context.